zcash_primitives/src/transaction/components/orchard.rs

```diff
@@ -311,50 +295,57 @@ impl<W: Write> WriteBurn<W> for OrchardZSA {
 /// Writes an [`orchard::Bundle`] in the appropriate transaction format.
 pub fn write_orchard_bundle<W: Write>(
     mut writer: W,
-    bundle: Option<&orchard::Bundle<Authorized, Amount, OrchardVanilla>>,
+    bundle: Option<&OrchardBundle<Authorized>>,
 ) -> io::Result<()> {
     if let Some(bundle) = &bundle {
-        Vector::write_nonempty(&mut writer, bundle.actions(), |w, a| {
-            write_action_without_auth(w, a)
-        })?;
-
-        writer.write_all(&[bundle.flags().to_byte()])?;
-        writer.write_all(&bundle.value_balance().to_i64_le_bytes())?;
-        writer.write_all(&bundle.anchor().to_bytes())?;
-        Vector::write(
-            &mut writer,
-            bundle.authorization().proof().as_ref(),
-            |w, b| w.write_u8(*b),
-        )?;
-        Array::write(
-            &mut writer,
-            bundle.actions().iter().map(|a| a.authorization()),
-            |w, auth| w.write_all(&<[u8; 64]>::from(*auth)),
-        )?;
-
-        writer.write_all(&<[u8; 64]>::from(
-            bundle.authorization().binding_signature(),
-        ))?;
+        match bundle {
+            OrchardBundle::OrchardVanilla(b) => write_orchard_vanilla_bundle(writer, b)?,
+            #[cfg(zcash_unstable = "nu6" /* TODO nu7 */ )]
+            OrchardBundle::OrchardZSA(b) => write_orchard_zsa_bundle(writer, b)?,
+        }
     } else {
         CompactSize::write(&mut writer, 0)?;
     }
 
     Ok(())
 }
 
+/// Writes an [`Bundle`] in the appropriate transaction format.
+pub fn write_orchard_vanilla_bundle<W: Write>(
```

Comment:
please make sure that the diff between upstream and us is just a rename from `write_v5_bundle()` and nothing more. (not the content of the function)

Reply:
I've synced them, there is only small difference in writing 0 if dundle is None, I do it one layer higher. Everything related to actual write of non-empty bundle is identical